src/python/pants/backend/docker/target_types.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.target import COMMON_TARGET_FIELDS, Dependencies, Sources, StringField, Target
+
+
+class DockerImageSources(Sources):
+    default = ("Dockerfile",)
+    expected_num_files = 1
+    help = "Name of the Dockerfile to use when building the Docker image."
+
+
+class DockerImageVersion(StringField):
+    alias = "version"
+    default = "latest"
+    help = "Image tag to apply to built images."
+
+
+class DockerDependencies(Dependencies):
+    pass
+
+
+class DockerBuildRoot(StringField):
```

Comment:
"build root" is a term with existing meaning in Pants (it means the directory containing the `./pants` script, which is the CWD when you run Pants - usually the repo root). So it's probably best not to overload that meaning here. Maybe "context root"?

Reply:
Right. Context root is perfect.